### PR TITLE
stm32l0: add support for external SPI flash

### PR DESCRIPTION
**Pull request by glennergeerts**

### CODE REVIEW

**Comment by dgarske** [comment]
Hi @glennergeerts , thank you for this submission adding STM32L0 external flash support. The changes look great.

In order to accept this we'll need a signed contributor agreement on file. Can you please email us directly at support@wolfssl.com and reference this PR? It would also be useful to hear about your project.

Thanks,
David Garske, wolfSSL

File: hal/spi/spi_drv_stm32.c [hunk 1]
```diff
@@ -90,6 +90,20 @@ static void spi1_pins_setup(void)
     SPI_PIO_AFL = reg | (SPI1_PIN_AF << ((SPI1_MOSI_PIN) * 4));
     reg = SPI_PIO_AFL & ~(0xf << ((SPI1_MISO_PIN) * 4));
     SPI_PIO_AFL = reg | (SPI1_PIN_AF << ((SPI1_MISO_PIN) * 4));
+
```
Comment (danielinux) [inline hal/spi/spi_drv_stm32.c:93]: These changes don't work on STM32F4. I suggest to move these settings to the STM32L0 hal if they are needed for SPI to work on that platform.

File: src/spi_flash.c [hunk 2]
```diff
@@ -179,13 +179,13 @@ uint16_t spi_flash_probe(void)
...
     spi_cs_off(SPI_CS_FLASH);
-    if (manuf == 0xBF)
+    if (manuf == 0xBF || manuf == 0xC2)
         chip_write_mode = SST_SINGLEBYTE;
     if (manuf == 0xEF)
         chip_write_mode = WB_WRITEPAGE;
 
 #ifndef READONLY
-    spi_cmd(EWSR);
+    //spi_cmd(EWSR); // TODO not supported on every flash, but necessary for some models?
```
Comment (danielinux) [inline src/spi_flash.c:188]: I have tested this change with SST25VF080B. The line is in fact not necessary and can be safely removed. Please remove the comment.

**Review by danielinux — CHANGES_REQUESTED** [review]
Please move pullup/output speed to hal_init(). The EWSR SPI command is indeed not necessary and can be removed.

**Comment by danielinux** [comment]
Merged! Thanks @glennergeerts

**Pull request merged**